src/ngraph/serializer.hpp

```diff
@@ -23,30 +23,38 @@
 
 namespace ngraph
 {
-    /// \brief Serialize a Function to a json string
+    /// \brief Serialize a Function to a json string or cpio
     /// \param func The Function to serialize
     /// \param indent If 0 then there is no formatting applied and the resulting string is the
     ///    most compact representation. If non-zero then the json string is formatted with the
     ///    indent level specified.
-    std::string serialize(std::shared_ptr<ngraph::Function> func, size_t indent = 0);
+    std::string serialize(std::shared_ptr<ngraph::Function> func,
```

Comment:
Better is to add separate `serialize_json` and `serialize_cpio` functions and let serialize continue doing what is used to do. There could be other kinds of serialization.

Reply:
Maybe pass some enum deciding on serialization kind then? Writing all of below three specialization overloads for each serialization kind seems to me over verbose.

Reply:
From my point of view the best solution will be creating common objective interface (abstract) to serialize and creation dedicated classes json_serializer and cpio_serializer which inherit the interface.
I've tried to implement this, but there are too many dependencies in ```serializer.cpp``` to do it without huge refactoring.

Implementation of separeted function for cpio and json cause 6 version of seralization method in ```serializer.hpp```

I've implemented version proposed by @arogowie-intel . What do you think about it?